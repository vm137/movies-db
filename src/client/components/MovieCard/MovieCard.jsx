import React from 'react';
import PropTypes from 'prop-types';
import './MovieCard.scss';

const MovieCard = ({ mv }) => (
  <div className="movie-card-wrapper">
    <div className="img-container">
      <img src={mv.poster_path} alt={mv.title} />
    </div>
    <div className="info-wrapper">
      <p className="movie-title">{mv.title}</p>
      <p className="movie-genres">{mv.genres.join(' & ')}</p>
      <p className="movie-year">{(mv.release_date).substring(0, 4)}</p>
    </div>
  </div>
);

MovieCard.propTypes = {
  mv: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default MovieCard;
