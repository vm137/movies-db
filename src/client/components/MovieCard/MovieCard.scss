@import '../constants/constants';

.movie-card-wrapper {
  float: left;
  width: 270px;
  height: 470px;
  margin: 0 0 20px 22px;
  background-color: #eaeaea;

  .img-container {
    width: 270px;
    height: 405px;

    img {
      width: 100%;
    }
  }

  .info-wrapper {
    position: relative;
    font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
    color: $TEXT_GREY;

    .movie-title {
      position: absolute;
      top: 12px;
      left: 12px;
      font-weight: bold;
    }

    .movie-genres {
      position: absolute;
      top: 38px;
      left: 12px;
      font-size: 14px;
    }

    .movie-year {
      position: absolute;
      top: 10px;
      right: 9px;
      padding: 4px;
      border: 1px solid $TEXT_GREY;
      font-size: 14px;
    }
  }

}
