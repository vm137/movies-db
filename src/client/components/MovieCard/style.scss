@import '../../constants/constants';

.movie-card-wrapper {
  float: left;
  width: 270px;
  height: 470px;
  margin: 0 0 20px 22px;
  background-color: #eaeaea;
  border: 1px lightgrey solid;
  box-sizing: content-box;
  cursor: pointer;

  .img-container {
    width: 270px;
    height: 405px;

    img {
      width: 100%;
    }
  }

  .info-wrapper {
    position: relative;
    font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
    color: $TEXT_GREY;

    .movie-title {
      position: absolute;
      width: 196px;
      top: 12px;
      left: 12px;
      font-weight: bold;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }

    .movie-genres {
      position: absolute;
      width: 250px;
      top: 38px;
      left: 12px;
      font-size: 13px;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }

    .movie-year {
      position: absolute;
      top: 8px;
      right: 8px;
      padding: 1px 8px;
      border: 1px solid $TEXT_GREY;
      font-size: 14px;
    }
  }

}
