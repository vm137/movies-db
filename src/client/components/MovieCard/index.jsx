import { connect } from 'react-redux';
import MovieCard from './MovieCard';
import { fetchSingleMovieAction } from '../../actions/actions';

const mapDispatchProps = {
  fetchSingleMovie: fetchSingleMovieAction,
};

export default connect(null, mapDispatchProps)(MovieCard);
