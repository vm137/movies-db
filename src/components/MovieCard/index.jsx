import { withRouter } from 'react-router-dom';
import MovieCard from './MovieCard';

export default withRouter(MovieCard);
