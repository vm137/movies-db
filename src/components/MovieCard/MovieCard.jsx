// @flow

import React, { PureComponent } from 'react';
import type { Movie } from '../../types';
import './style.scss';

type DefaultProps = {| history: {} |};
type Props = {
  ...DefaultProps,
  movie: Movie,
  history: Object
};

export default class MovieCard extends PureComponent<Props> {
  constructor(props: Props) {
    super(props);
    (this: Object).handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    const { movie, history } = this.props;
    history.push(`/film/${movie.id}`);
  }

  render() {
    const { movie } = this.props;

    return (
      <div onClick={this.handleClick} role="button" tabIndex={0} onKeyPress={() => {}} className="movie-card-wrapper">
        <div className="img-container">
          <img src={movie.poster_path} alt={movie.title} />
        </div>
        <div className="info-wrapper">
          <p className="movie-title">{movie.title}</p>
          <p className="movie-genres">{movie.genres.join(' & ')}</p>
          <p className="movie-year">{(movie.release_date).substring(0, 4)}</p>
        </div>
      </div>
    );
  }
}
