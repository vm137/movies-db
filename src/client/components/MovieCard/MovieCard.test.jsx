import { shallow, render } from 'enzyme';
import React from 'react';
import testData from './__testdata__/testMovie.json';
import MovieCard from './MovieCard';

describe('<MovieCard />', () => {
  const onClick = () => {};

  it('renders matching snapshot', () => {
    const wrapper = shallow(<MovieCard mv={testData} onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('find .movie-title class', () => {
    const wrapper = shallow(<MovieCard mv={testData} onClick={onClick} />);
    expect(wrapper.find('.movie-title').length).toBe(1);
  });

  it('find genre from array of genres', () => {
    const wrapper = shallow(<MovieCard mv={testData} onClick={onClick} />);
    expect(wrapper.find('.movie-genres').text()).toBe('Western');
  });

  it('check handleClick() method', () => {
    const onClickJest = jest.fn();
    const wrapper = shallow(<MovieCard mv={testData} onClick={onClickJest} />);
    const divWrapper = wrapper.find('.movie-card-wrapper');
    expect(divWrapper.length).toBe(1);
    divWrapper.simulate('click');
    expect(onClickJest.mock.calls.length).toEqual(1);
  });
});
