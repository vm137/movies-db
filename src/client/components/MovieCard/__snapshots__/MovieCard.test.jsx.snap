// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders matching snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "movie-card-wrapper",
      "role": "button",
      "tabindex": "0",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "img-container",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "alt": "Duck, You Sucker",
              "src": "https://image.tmdb.org/t/p/w500/nQkr1yi87pX1d3lQVTrBEWlCpT1.jpg",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "src": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "info-wrapper",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "movie-title",
              },
              "children": Array [
                Object {
                  "data": "Duck, You Sucker",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "movie-genres",
                },
                "children": Array [
                  Object {
                    "data": "Western",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "movie-year",
                  },
                  "children": Array [
                    Object {
                      "data": "1971",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "movie-genres",
              },
              "children": Array [
                Object {
                  "data": "Western",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "movie-year",
                },
                "children": Array [
                  Object {
                    "data": "1971",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "movie-title",
                },
                "children": Array [
                  Object {
                    "data": "Duck, You Sucker",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "movie-year",
              },
              "children": Array [
                Object {
                  "data": "1971",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "movie-genres",
                },
                "children": Array [
                  Object {
                    "data": "Western",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "movie-title",
                  },
                  "children": Array [
                    Object {
                      "data": "Duck, You Sucker",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "info-wrapper",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "movie-title",
            },
            "children": Array [
              Object {
                "data": "Duck, You Sucker",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "movie-genres",
              },
              "children": Array [
                Object {
                  "data": "Western",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "movie-year",
                },
                "children": Array [
                  Object {
                    "data": "1971",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "movie-genres",
            },
            "children": Array [
              Object {
                "data": "Western",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "movie-year",
              },
              "children": Array [
                Object {
                  "data": "1971",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "movie-title",
              },
              "children": Array [
                Object {
                  "data": "Duck, You Sucker",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "movie-year",
            },
            "children": Array [
              Object {
                "data": "1971",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "movie-genres",
              },
              "children": Array [
                Object {
                  "data": "Western",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "movie-title",
                },
                "children": Array [
                  Object {
                    "data": "Duck, You Sucker",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "img-container",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "alt": "Duck, You Sucker",
                "src": "https://image.tmdb.org/t/p/w500/nQkr1yi87pX1d3lQVTrBEWlCpT1.jpg",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "alt": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "alt": undefined,
                "src": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "role": undefined,
      "tabindex": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "role": undefined,
      "tabindex": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
